pkg/scheduler/plugins/coscheduling/core/core.go

```diff
@@ -110,15 +113,43 @@ func NewPodGroupManager(
 		DeleteFunc: gangCache.onPodGroupDelete,
 	}
 	frameworkexthelper.ForceSyncFromInformer(context.TODO().Done(), pgSharedInformerFactory, pgInformer.Informer(), podGroupEventHandler)
+	pgMgr.CleanOrphanGang()
 
 	podEventHandler := cache.ResourceEventHandlerFuncs{
 		AddFunc:    gangCache.onPodAdd,
 		DeleteFunc: gangCache.onPodDelete,
 	}
 	frameworkexthelper.ForceSyncFromInformer(context.TODO().Done(), sharedInformerFactory, podInformer.Informer(), podEventHandler)
+
 	return pgMgr
 }
 
+func (pgMgr *PodGroupManager) CleanOrphanGang() {
```

Comment:
Thanks for the comment. : )
I delete the pod group directly here because we will doing the same things in `onPodDelete()`. By the current design of gang scheduling, if a pod group is created by pod annotation, we will delete it once we find its children have all been deleted. 
In addition, if the pod group is create by the pod annotation and the gang object doesn't in scheduler, the lifecycle of the pod group has ended because its owner, the pod who create it, has been deleted. In this situation, is it necessary to hold the pod group object in api server?

Reply:
I think there should be only one PodGroup GC implementation, rather than similar implementations appearing in many places. Existing delete PodGroup in Pod Delete EventHandler is not a good way. The latency of a request to APIServer is at least ten or even hundreds of milliseconds, which will affect the processing efficiency of EventHandler. This should be optimized for asynchronous processing. For some scenarios where other components create PodGroups by themselves but miss, we should be responsible for GC, but not limited to only in the scheduler initialization phase, but should be in the entire scheduler running cycle.

Reply:
I prefer put the logic in backround loop check like GC implementation.  the logic itself seems ok.